fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/tree/ScalarOperatorsReuse.java

```diff
@@ -438,7 +439,8 @@ public static Projection rewriteProjectionOrLambdaExpr(Projection projection, Co
             // Apply to normalize rule to eliminate invalid ColumnRef usage for in-predicate
             com.starrocks.sql.optimizer.rewrite.ScalarOperatorRewriter rewriter =
                     new com.starrocks.sql.optimizer.rewrite.ScalarOperatorRewriter();
-            List<ScalarOperatorRewriteRule> rules = Collections.singletonList(new NormalizePredicateRule());
+            List<ScalarOperatorRewriteRule> rules =
+                    ImmutableList.of(new NormalizePredicateRule(), new ReduceCastRule());
```

Comment:
Why need reduce cast here?

Reply:
can be reduce cast in some in-predicate case